src/protections/WordList.ts

```diff
@@ -17,23 +17,17 @@ limitations under the License.
 import { Protection } from "./IProtection";
 import { Mjolnir } from "../Mjolnir";
 import { LogLevel, LogService } from "matrix-bot-sdk";
-import config from "../config";
 import { isTrueJoinEvent } from "../utils";
 
 export class WordList extends Protection {
 
     settings = {};
 
     private justJoined: { [roomId: string]: { [username: string]: Date} } = {};
-    private badWords: RegExp;
+    private badWords?: RegExp;
```

Comment:
So this becomes lazily initialized?

Fair enough, but could you document this?

Reply:
Ok serious JavaScript question, if something is to be initialized later on, why does it make sense to have the initial value be undefined? surely this would suggest this property does not exist? Rather than if it were null, which suggests it does exist but hasn't been initialized yet. I've noticed the practice creeping in on other Mjolnir PRs and while it isn't a big thing, I'm interested in the explanation.

Reply:
I'd say that either choice is good as long as we pick one, stick to it and document it. But now that you mention it, I agree that that `null` sounds clearer. I suspect that we use `undefined` simply because that makes it shorter to write, but that's a bad reason to do things.

Reply:
So not `null`?